apps/ejabberd/src/mongoose_acc.erl

```diff
@@ -63,9 +63,10 @@ from_kv(K, V) ->
 
 -spec from_element(xmlel()) -> t().
 from_element(El) ->
-    #xmlel{name = Name, attrs = Attrs} = El,
-    Type = xml:get_attr_s(<<"type">>, Attrs),
-    #{element => El, mongoose_acc => true, name => Name, attrs => Attrs, type => Type}.
+    #xmlel{name = Name, attrs = Attrs, children = Children} = El,
+    Type = exml_query:attr(El, <<"type">>, undefined),
+    Acc = #{element => El, mongoose_acc => true, name => Name, attrs => Attrs, type => Type},
+    read_children(Acc, Children).
```

Comment:
I'm reluctant to inspect every stanza so deeply. Do we really need to do this for every stanza or maybe there is only limited number of places where it's needed?